spec/puppet-lint/plugins/check_documentation/advanced_doc_comments_rules.json

```diff
@@ -0,0 +1,13 @@
+{
+    "summary-exists": {
+        "check": "@summary"
+    },
+    "summary-is-indented": {
+        "comment-condition": "@summary",
+        "check": "@summary\\n  .+"
+    },
+    "params-are-documented": {
+        "condition": "(class|define) [^ ]+ \\([^\\)]+\\)",
+        "check": "@param"
+    }
```

Comment:
We do have https://github.com/voxpupuli/puppet-lint-param-docs already. I'm generally not a fan of multiple ways to do the same test. That plugin actually tells you which params are undocumented, which is much stronger than this implementation.

Reply:
Yes, this is actually just a test for the spec, not intended to replace the other one.

Reply:
Do you want me to change this to something else because the file is mentioned in the README and may suggest that it would replace param-docs?

Reply:
To be clear: I'm not a maintainer on this repo. I just have a personal interest.

I've also been thinking about being stricter on strings output. One thing I've been thinking about is that it may be good to generate the generated strings (using the JSON output) and somehow verify that. And if there are `@param` blocks for parameters that don't exist, that's something puppet-strings itself could already detect and warn about.

If we get to that point, the only example I see here are about formatting of tags. Perhaps `puppet-strings` would actually be a better place to implement those. Any thoughts?

Reply:
We have some more advanced requests here. For example, @summary should be on single line and the actual summary indented on the next. That's why I chose this general approach.

> I've also been thinking about being stricter on strings output. One thing I've been thinking about is that it may be good to generate the generated strings (using the JSON output) and somehow verify that. And if there are `@param` blocks for parameters that don't exist, that's something puppet-strings itself could already detect and warn about.

What do you mean here?